predict_bias.py

```diff
@@ -71,45 +78,49 @@ def read_input(): # needs clean up with the proprocessor and dataset
     # SFC input
     sfcs = []
     for var in sfc_vars:
-        sfcs.append(file_s[var].values) # shape(1,32,64)
+        sfcs.append(file_s[var].values) 
 
-    sfcs.append(lats_m[None,]) #22
-    sfcs.append(lons_sin[None,]) #23
-    sfcs.append(lons_cos[None,]) #24
-    sfcs.append(np.ones((1,nlat,nlon))*date_in[:,None,None])
+    sfcs.append(lons_m[None,]) # added for consistency with preprocessor
+    sfcs.append(lats_m[None,]) 
+    sfcs.append(lons_sin[None,]) 
+    sfcs.append(lons_cos[None,]) 
+    sfcs.append(np.ones((1,nlat,nlon))*date_in[:,None,None]) 
 
     # Combine the inputs
-    nbc = 21
-    slice_f06 = slice(0,509)
-    slice_sfc = [14,15,16,17,18,19,20]+list(range(nbc,nbc+7)) 
-    #slice_sfc = list(range(0,nbc))+list(range(nbc+1,nbc+8))
+    #nbc = len(sfc_vars) #21 
+    slice_f06 = slice(0,509) # 509 = 4*127 + 1 (127=nlev, 4=num  3d variables and one sfc variable as input)
+    slice_sfc = slice(509,526) 
     
     ins = torch.cat([torch.from_numpy(np.concatenate(vals_f, axis=0)[None,]), 
-                     torch.from_numpy(np.concatenate(sfcs, axis=0)[None,slice_sfc])], 1).float()
+                     torch.from_numpy(np.concatenate(sfcs, axis=0)[None,])], 1).float()
+    print(np.shape(torch.from_numpy(np.concatenate(vals_f, axis=0)[None,])))
+    print(np.shape(torch.from_numpy(np.concatenate(sfcs, axis=0)[None,])))
     # Prepare Normalizing mean and std
-    ddd='/scratch2/NCEPDEV/stmp1/Tse-chun.Chen/anal_inc/npys/ifs'
-    mean_f06 = torch.from_numpy(np.load(ddd+'_f06_ranl_{}_mean_1d.npy'.format(trunc))[slice_f06])
+    ddd='/scratch2/BMC/gsienkf/Sergey.Frolov/fromStefan/npys_sergey2/ifs'
```

Comment:
Should we move this line above function declaration so it is specified once for this script?

Reply:
Sure I'll do that